sgx/pelz_enclave.edl

```diff
@@ -4,77 +4,41 @@ enclave {
 	from "kmyth_enclave.edl" import *;
         include "aes_keywrap_3394nopad.h"
 	include "pelz_request_handler.h"
+	include "common_table.h"
 	include "key_table.h"
 	include "server_table.h"
 	include "pelz_io.h"
 	include "charbuf.h"
         include "util.h"
 	trusted {
-/**
- * <pre>
- * This function initializes a key table for lookup.
- * <pre>
- *
- * @param[in] key_table The pointer for key table to be initialized
- * @param[out] key_table The initialized key table
- *
- * @return 0 on success, 1 on error
- */
-public int key_table_init(void);
 
 /**
  * <pre>
- * This function destroys the key table.
+ * This function destroys the table specified by type.
  * <pre>
  *
- * @param[in] key_table The key table to be destroyed
+ * @param[in] type The type of table to be destroyed (ex: key_table or server_table)
  *
  * @return 0 on success, 1 on error
  */
-public int key_table_destroy(void);
+public int table_destroy(int type);
 
 /**
  * <pre>
- * This function to delete values in hash table based on location in key_id.
+ * This function deletes a value in hash table based on location in id.
  * <pre>
  *
- * @param[in] key_id.chars Key identifier assumed to be null terminated
- * @param[in] key_id.len The length of the key identifier
- * @param[in] key_table The key table that the key needs to be deleted from
- * @param[out] key_table The key table without the deleted key
- *
- * @return 0 on success, 1 on error
- */
-public int key_table_delete(charbuf key_id);
-
-/**
- * <pre>
- * This function destroys the server table.
- * <pre>
- *
- * @param[in] server_table The server table to be destroyed
- *
- * @return 0 on success, 1 on error
- */
-public int server_table_destroy(void);
-
-/**
- * <pre>
- * This function to delete values in hash table based on location in key_id.
- * <pre>
- *
- * @param[in] server_id.chars Server identifier assumed to be null terminated
- * @param[in] server_id.len The length of the server identifier
- * @param[in] server_table The server table that the cert needs to be deleted from
- * @param[out] server_table The server table without the deleted cert
+ * @param[in] type The table type that the id and value needs to be deleted from
+ * @param[in] id.chars Identifier assumed to be null terminated
```

Comment:
If we pass in the identifier length why do we bother assuming the identifier is null terminated?